ng/js/controllers/new_posting.js

```diff
@@ -0,0 +1,270 @@
+require('angular').module('ays')
+  .controller('CreatePostingController', function ($scope, $sails, $timeout, $state) {
+    $scope.categoryOptions = [];
+    $scope.categoryDescription = '';
+    $scope.titleErrorMessages = [];
+    $scope.priceErrorMessages = [];
+    $scope.descriptionErrorMessages = [];
+    $scope.categoryErrorMessages = [];
+
+    $scope.allInputValidated = false;
+    $scope.titleValidated = false;
+    $scope.pricesValidated = false;
+    $scope.descriptionValidated = false;
+    $scope.categoryValidated = false;
+    $scope.isSubmitting = false;
+
+    // Populate category options based on what server has
+    $sails.get('/api/category')
+      .then(function (response) {
+        var i = 0;
+        for (i = 0; i < response.data.length; i++) {
+          $scope.categoryOptions.push(response.data[i].description);
+        }
+      });
+
+    $scope.newPosting = {
+      sellerID: '',
+      id: '',
+      title: '',
+      categoryID: '',
+      image: null,
+      description: '',
+      price: {
+        minimum: '',
+        maximum: '',
+      },
+      date: '',
+      tags: '',
+      skills: '',
+      active: false,
+    };
+
+    // Simple helper function that resets the newposting object to default values.
+    function getDisplayObject() {
+      return {
+        sellerID: '',
+        id: '',
+        title: '',
+        categoryID: '',
+        image: null,
+        description: '',
+        price: {
+          minimum: '',
+          maximum: '',
+        },
+        date: '',
+        tags: '',
+        skills: '',
+        active: false,
+      };
+    }
+
+    // Submit the form. Only gets called after validation occurs.
+    $scope.submitForm = function () {
+      $scope.newPosting.categoryID = $scope.categoryOptions.indexOf($scope.categoryDescription);
```

Comment:
The list 'categoryOptions' is populated in the UI based on the order its retrieved from the server. This makes the data that's POSTed to /api/posting contain a 'categoryID' field that accurately reflects the list on the server itself. That should make this okay, right?